src/vsr.zig

```diff
@@ -687,7 +714,7 @@ pub const Timeout = struct {
 
     /// It's important to check that when fired() is acted on that the timeout is stopped/started,
     /// otherwise further ticks around the event loop may trigger a thundering herd of messages.
-    pub fn fired(self: *Timeout) bool {
+    pub fn fired(self: *const Timeout) bool {
```

Comment:
Would it be correct to check `.fired()` two times in a row? The code looks like it wants to set some flag or something for fired timers...

Reply:
:thinking: If it was `.fire()` I would agree, but since it is `.fired()` it seems ok to call it multiple times in the same tick.

Reply:
Yeah, so we can do either of: 

* keep name, change to `*const`
* keep `*Timeout`, change name to `fire`, add assert

The comment above makes me think that maybe the second one is the right approach. 

:thinking: or we can even have both functions, asserting`.fire()` and  const `.is_firing()`, with the expectation that most call-sites want `.fire`, but some specific ones need `is_firing()` (and guard against thundering herd in some way. )

Reply:
Also perhaps out of scope for the PR, as we don't have assert today, so it's not clear at a glance if "call multiple times in a tick" is happenning or not in practiec.

Reply:
Since `fire()`/`fired()` (either one) _must_ be called at _every_ tick (otherwise we [panic](https://github.com/tigerbeetledb/tigerbeetle/blob/bfbcd126353d0161b8fbcd14260759c39709071a/src/vsr.zig#L693-L696)), it seems like `tick()` could return a "did it fire" boolean. i.e. instead of:

```zig
            self.ping_timeout.tick();
...
            if (self.ping_timeout.fired()) self.on_ping_timeout();
```
it would be


```zig
            if (self.ping_timeout.tick()) self.on_ping_timeout();
```

Reply:
Ah, we _only_ ever call `.tick` and `.fire()` in that small section of code, this is so much better than what I imagined happening. I think that just changing to `*const` is best then. I _like_ the current setup where we first tick all timers, and then check each one for `fired`, because it's more order-independent: logically, we advance all timers at the same time, there no transitionary states where some timers were advanced, and some were not.